src/main/java/com/work/management/service/log/impl/LoggingServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.work.management.service.log.impl;
+
+import com.google.common.collect.ImmutableMap;
+import com.work.management.service.log.LoggingService;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+final class LoggingServiceImpl implements LoggingService {
+
+  private static final Logger logger = LoggerFactory.getLogger(LoggingServiceImpl.class);
+
+  @Override
+  public void logRequest(HttpServletRequest httpServletRequest, Object body) {
+    StringBuilder stringBuilder = new StringBuilder();
+    ImmutableMap<String, String> parameters = buildParametersMap(httpServletRequest);
+
+    stringBuilder.append("REQUEST ").append("method=[").append(httpServletRequest.getMethod())
+        .append("] ").append("path=[").append(httpServletRequest.getRequestURI()).append("] ")
+        .append("headers=[").append(buildHeadersMap(httpServletRequest)).append("] ");
+
+    if (!parameters.isEmpty()) {
+      stringBuilder.append("parameters=[").append(parameters).append("] ");
+    }
+
+    if (!Objects.isNull(body)) {
+      stringBuilder.append("body=[" + body + "]");
```

Comment:
change to: .append("body=[").append(body).append("]");
The implementation you are using will create unnecessary strings in string pool, which can cause heapOutOfMemory issue.

Reply:
Oops I am wrong: https://stackoverflow.com/questions/17487373/stringbuilder-uses-more-memory-when-append-is-used
 but yeah we should keep the same code everywhere for consistency.